### PR TITLE
Support Shoot control plane HA in KEB

### PR DESCRIPTION
**Pull request by ebensom**
**Description**

Changes proposed in this pull request:

- Support configuring Shoot control plane HA (failure tolerance level) for enterprise plans
- Propagate control plane failure tolerance configuration to provisioner during provisioning

**Related issue(s)**

### CODE REVIEW

**Review by wozniakjan — COMMENTED** [review]
ever since https://github.com/kyma-project/test-infra/pull/5776, any change in KEB go mod requires KCP `go mod tidy` too.. (more about that in https://github.com/kyma-project/control-plane/issues/1929)

**Comment by ebensom** [comment]
> ever since [kyma-project/test-infra#5776](https://github.com/kyma-project/test-infra/pull/5776), any change in KEB go mod requires KCP `go mod tidy` too.. (more about that in #1929)

Thanks a lot, resolved now.

File: resources/kcp/charts/kyma-environment-broker/templates/deployment.yaml [hunk 1]
```diff
@@ -334,6 +334,8 @@ spec:
               value: "{{ .Values.gardener.autoUpdateMachineImageVersion }}"
             - name: APP_PROVISIONER_MULTI_ZONE_CLUSTER
               value: "{{ .Values.gardener.multiZoneCluster }}"
+            - name: APP_PROVISIONER_CONTROL_PLANE_FAILURE_TOLERANCE
+              value: "{{ .Values.gardener.controlPlaneFailureTolerance }}"
```
Comment (wozniakjan) [inline resources/kcp/charts/kyma-environment-broker/templates/deployment.yaml:338]: nit: given this is configured in KEB for all requests the same, does it make sense to keep passing this explicitly from KEB in every gardener payload to provisioner? could be simpler to configure once in the provisioner and save on the payloads.

Comment (ebensom) [reply resources/kcp/charts/kyma-environment-broker/templates/deployment.yaml:338]: This is not passed in every payload from KEB to provisioner, it is only passed for shoots of enterprise plans, not for trial, free and azure_lite.
provisioner doesn't have the knowledge about "SKR plans", and wouldn't know to which cluster's should it apply the configured failure tolerance configuration.

**Review by wozniakjan — APPROVED** [review]

**Pull request merged**